### ISSUE TITLE
Usability feedback: Error Messages

### ISSUE THREAD

**Issue by Sparrow0hawk** [acteng/update-your-capital-schemes#140]
# Usability feedback

## Error Messages (Usability)

Error messages for date component does not follow GOV.UK Design System guidance.

### Reference:
Usability & GOV.UK Design System requirement

## Issue ID: DAC_Error_Messages_Usability_01

URL: https://test.update-your-capital-schemes.activetravelengland.gov.uk/schemes/64/milestones
Page title: 'Error: Update your capital schemes - Active Travel England - GOV.UK'
H1 heading: 'Change milestone dates'
Journey: 1.5

Screenshot:
![image](https://github.com/user-attachments/assets/17163053-4ed5-46a7-894f-7d16edaf1f54)

On the 'Change milestone dates' page, when the 'year' field is missing, a generic error message is provided, rather than the specific error guidance as found in [GOV.UK Design System: Date input (Error messages)](https://design-system.service.gov.uk/components/date-input/#error-messages).

### Current code ref(s):
#main-content > div > div > form > div:nth-child(3) > div:nth-child(1) > div > fieldset
```html
<fieldset class="govuk-fieldset" role="group" aria-describedby="date-hint feasibility_design_completed_planned-error">
   <legend class="govuk-fieldset__legend">
      Planned date
   </legend>
   <p id="feasibility_design_completed_planned-error" class="govuk-error-message">
      <span class="govuk-visually-hidden">
         Error:
      </span>
      Enter a feasibility design completed planned date
   </p>
   […]
</fieldset>
```

### Examples of additional instances:

Additional instances of this issue may exist on other pages throughout the website; wherever this issue occurs, they too will need to be resolved.

### Solution:

Ensure that the comprehensive error message guidance, for the date component, found at [GOV.UK Design System: Date input (Error messages)](https://design-system.service.gov.uk/components/date-input/#error-messages) is followed.

In this instance, as detailed in [GOV.UK Design System: Date input (If the date is incomplete)](https://design-system.service.gov.uk/components/date-input/#if-the-date-is-incomplete), the guidance states that only the fields which are missing or incomplete should be highlighted (and see issue [DAC_Error_Links_Usability_01](https://github.com/acteng/schemes/issues/139) regarding linking the error summary link to the first field that is in error).

The error message should follow the format of '[Form Field] must include a [Missing information]'. Such as, in this instance, 'Feasibility design completed planned date must include a year'.
